TypeCobol.Analysis.Test/PerformanceWithCfg.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TypeCobol.Compiler;
+using TypeCobol.Compiler.Diagnostics;
+using TypeCobol.Compiler.Directives;
+using TypeCobol.Compiler.Text;
+using TypeCobol.Test.Compiler.Parser;
+using TypeCobol.Test.Utils;
+
+namespace TypeCobol.Analysis.Test
+{
+    [TestClass]
+    [Ignore]
+    public class PerformanceWithCfg : TypeCobol.Test.Parser.Performance.Performance
+    {
+        /// <summary>
+        /// AnalyzerProvider for test.
+        /// </summary>
+        /// <returns></returns>
+        private static CompositeAnalyzerProvider MyAnalyzerProvider()
+        {
+            //Add analyzers
+            var analyzerProvider = new CompositeAnalyzerProvider();
+            //CFG/DFA
+            const string cfgDfaId = "cfg-dfa";
+            analyzerProvider.AddActivator((o, t) => CfgDfaAnalyzerFactory.CreateCfgAnalyzer(cfgDfaId, CfgBuildingMode.Standard));
+            return analyzerProvider;
+        }
+
+        /// <summary>
+        /// Method for parsing a document.
+        /// </summary>
+        /// <param name="fullPath"></param>
+        /// <param name="options"></param>
+        /// <param name="format"></param>
+        /// <param name="copiesFolder"></param>
+        /// <returns></returns>
+        protected override TypeCobol.Parser ParseDocument(string fullPath, TypeCobolOptions options, TypeCobol.Compiler.DocumentFormat format, string[] copiesFolder)
+        {
+            var document = new TypeCobol.Parser();
+            document.Init(fullPath, options, format, copiesFolder, MyAnalyzerProvider());
+            document.Parse(fullPath);
+            return document;
+        }
+
+        protected override CompilationProject CreateCompilationProject(string projectName, string rootDirectory, string[] fileExtensions, DocumentFormat documentFormat, TypeCobolOptions compilationOptions, IAnalyzerProvider analyzerProvider)
+        {
+            return base.CreateCompilationProject(projectName, rootDirectory, fileExtensions, documentFormat, compilationOptions, MyAnalyzerProvider());
+        }
```

Comment:
Ok so it looks like `ParseDocument` and `CreateCompilationProject` are respectively used for FullParsing and Incremental. You can improve factorization by removing these 2 methods (as they only differ at the type of analyzer used in derived classes) and replace them with something like :
```csharp
protected abstract void AddAnalyzers(CompositeAnalyzerProvider analyzerProvider);
```
in the base class `Performance`. No need to change anything from original code except to create a `CompositeAnalyzerProvider`, call `AddAnalyzer` and pass the instance either on the `Parser.Init` or into the `CompilationProject`.

Reply:
Done in 09d4586